manifests/tlog.pp

```diff
@@ -1,9 +1,13 @@
 # @summary Setup and configure tlog
 #
+# @param conf
+#   tlog config for /etc/tlog/tlog-rec-session.conf
+#
 # @example
 #   include profile_session_log::tlog
-class profile_session_log::tlog {
-
+class profile_session_log::tlog (
+  String $conf,
+) {
   $enabled = lookup("${module_name}::enable_session_log", Boolean)
   $required_pkgs = lookup("${module_name}::install::required_pkgs", Array)
```

Comment:
I wonder if it would make more sense to set these 2 parameters in the base class, then let the subclasses reference these parameters from there. I think that subclasses automatically know about variables scoped from their parent classes. See https://www.puppet.com/docs/puppet/7/lang_scope

Reply:
Yeah that looks better. I've updated the params to be in the parent and removed the lookups.